controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -130,6 +132,56 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 	return response, nil
 }
 
+// ReconcileEtcdMembers iterates over all ETCD members and finds members with nodes that do not belong to the workload cluster.
+// If there are any such members, it deletes those members and remove nodes from kubeadmconfig so that Kubeadm does not try to run etcd-health check over them.
+func (w *Workload) ReconcileEtcdMembers(ctx context.Context) error {
+	controlPlaneNodes, err := w.getControlPlaneNodes(ctx)
+	if err != nil {
+		return err
+	}
+
+	errList := []error{}
+	for _, node := range controlPlaneNodes.Items {
```

Comment:
I still don't think we need this outer loop. We want to loop through every etcd member, meaning for a 3 node cluster cluster we only need to call `Members` once, and then loop through the 3 members, breaking once we find the matching one. We could use `etcdClientGenerator.forLeader`.

Reply:
We enter this reconcile function when the etcd is not healthy. There may be no leader at that point. Having these extra iterations make things safer here.

Reply:
When https://github.com/kubernetes-sigs/cluster-api/pull/2888 is in, we can drop that loop. I am opening another issue to track this.